apps/ryal_core/lib/ryal/web/models/payment_method.ex

```diff
@@ -2,14 +2,72 @@ defmodule Ryal.PaymentMethod do
   @moduledoc """
   A standard adapter to multiple payment methods.
 
-  TODO: Payment Method documentation as this model becomes more fledged out.
+  You can specify which payment you'd like to use via the `type` field and
+  placing the data of a payment in the `data` field.
+
+  The `data` field uses PostgreSQL's JSONB column to store the dynamic
+  information. We use validations and whatnot at the application level to ensure
+  the data is consistent. (Although, it would be nice is we could add
+  constraints to PG later on.)
+
+  ## Example
+
+      iex> Ryal.PaymentMethod.changeset(%Ryal.PaymentMethod{}, %{
+        type: "credit_card",
+        data: %{
+          name: "Bobby Orr",
+          number: "4242 4242 4242 4242",
+          month: "03",
+          year: "2048",
+          cvc: "004"
+        }
+      })
+
+      #Ecto.Changeset<action: nil,
+       changes: %{data: #Ecto.Changeset<action: :insert,
+          changes: %{cvc: "123", month: "03", name: "Bobby Orr",
+            number: "4242 4242 4242", year: "2048"}, errors: [],
+          data: #Ryal.PaymentMethod.CreditCard<>, valid?: true>,
+          type: "credit_card"},
+       errors: [], data: #Ryal.PaymentMethod<>, valid?: true>
   """
 
   use Ryal.Web, :model
 
   schema "ryal_payment_methods" do
+    field :type, :string
+
+    embeds_one :data, Ryal.PaymentMethod.Proxy
+
     has_many :payment_method_gateways, Ryal.PaymentMethodGateway
 
     belongs_to :user, Ryal.user_module()
   end
+
+  @required_fields ~w(type user_id)a
```

Comment:
Now that I think about it, it's kinda confusing/smart. I wonder if there's a shorthand _just_ for atoms like how ruby has `%i()`.